pkg/cmd/server/origin/asset.go

```diff
@@ -38,50 +38,17 @@ func (c *AssetConfig) WithAssets(handler http.Handler) (http.Handler, error) {
 		return handler, nil
 	}
 
-	publicURL, err := url.Parse(c.Options.PublicURL)
-	if err != nil {
-		return nil, err
-	}
-
-	mux := http.NewServeMux()
-	err = c.addHandlers(mux)
-	if err != nil {
-		return nil, err
-	}
-
-	if !strings.HasSuffix(publicURL.Path, "/") {
-		publicURL.Path = publicURL.Path + "/"
-	}
-
-	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		if strings.HasPrefix(req.URL.Path + "/", publicURL.Path) {
-			mux.ServeHTTP(w, req)
-		} else {
-			handler.ServeHTTP(w, req)
-		}
-	}), nil
+	return c.addHandlers(handler)
 }
 
 // Run starts an http server for the static assets listening on the configured
 // bind address
 func (c *AssetConfig) Run() {
-	publicURL, err := url.Parse(c.Options.PublicURL)
-	if err != nil {
-		glog.Fatal(err)
-	}
-
-	mux := http.NewServeMux()
-	err = c.addHandlers(mux)
+	mux, err := c.addHandlers(nil)
```

Comment:
Won't that make any path that doesn't match the asset handler redirect to the asset url, instead of falling through to the next handler?